libff/algebra/curves/profile/algebra_groups_profile.cpp

```diff
@@ -111,6 +111,79 @@ template<typename GroupT> bool profile_group_membership()
     return true;
 }
 
+template<typename ppT> bool profile_pairing_e_over_e()
+{
+    constexpr size_t NUM_ITERATIONS = ADD_NUM_DIFFERENT_ELEMENTS;
+
+    // Mock BLS signature verification
+
+    const Fr<ppT> sk = Fr<ppT>::random_element();
+    G2<ppT> pk = sk * G2<ppT>::one();
+    pk.to_affine_coordinates();
+
+    // Messages to sign
+    std::vector<Fr<ppT>> messages;
+    messages.reserve(NUM_ITERATIONS);
+    for (size_t i = 0; i < NUM_ITERATIONS; ++i) {
+        messages.push_back(Fr<ppT>::random_element());
+    }
+
+    // Signatures. Every other one is invalid.
+    std::vector<G1<ppT>> signatures;
+    signatures.reserve(NUM_ITERATIONS);
+    for (size_t i = 0; i < NUM_ITERATIONS; ++i) {
+        if (i & 1) {
+            signatures.push_back(sk * messages[i] * G1<ppT>::one());
+        } else {
+            signatures.push_back(messages[i] * G1<ppT>::one());
+        }
+        signatures.back().to_affine_coordinates();
+    }
+
+    // Precompute the parts we expect to be known up-front.
+    const G2_precomp<ppT> id_precomp = ppT::precompute_G2(-G2<ppT>::one());
+    const G2_precomp<ppT> pk_precomp = ppT::precompute_G2(pk);
+
+    enter_block("pairing_e_over_e");
+
+    // NOTE: the pairing implementations call enter_block and leave_block in
+    // several places, generating a lot of output, which slows down the
+    // performance a lot. Hence we suppress the profiling here, and re-enable
+    // it before the end of the block.
+
+    libff::inhibit_profiling_info = true;
+    libff::inhibit_profiling_counters = true;
+
+    for (size_t i = 0; i < NUM_ITERATIONS; ++i) {
+        const G1_precomp<ppT> sig_prec = ppT::precompute_G1(signatures[i]);
+        const G1_precomp<ppT> H_precomp =
+            ppT::precompute_G1(messages[i] * G1<ppT>::one());
+
+        const Fqk<ppT> result =
+            ppT::final_exponentiation(ppT::double_miller_loop(
+                sig_prec, id_precomp, H_precomp, pk_precomp));
+
+        if (i & 1) {
+            if (Fqk<ppT>::one() != result) {
```

Comment:
`if (Fqk<ppT>::one() != result) {` -> `if (result != Fqk<ppT>::one()) {` for consistency